bellows/zigbee/zcl/__init__.py

```diff
@@ -192,13 +192,22 @@ def write_attributes(self, attributes):
         for attrid, value in attributes.items():
             if isinstance(attrid, str):
                 attrid = self._attridx[attrid]
+            if attrid not in self.attributes:
+                self.error("%d is not a valid attribute id" % attrid)
+                continue
+
             a = foundation.Attribute()
             a.attrid = t.uint16_t(attrid)
             a.value = foundation.TypeValue()
-            python_type = self.attributes[attrid][1]
-            a.value.type = t.uint8_t(foundation.DATA_TYPE_IDX[python_type])
-            a.value.value = python_type(value)
-            args.append(a)
+
+            try:
+                python_type = self.attributes[attrid][1]
+                a.value.type = t.uint8_t(foundation.DATA_TYPE_IDX[python_type])
+                a.value.value = python_type(value)
+                args.append(a)
+            except ValueError as e:
+                self.error(e)
```

Comment:
I'm guessing the `str(msg)` in `log` is to handle this. I think you should either stringify the exception here, or add an exception logging method to the `LocalLogMixin`, that matches the logging module's exception... Pretend that the methods are strongly typed and need a string for `msg`.

I think ...